src/reason-parser/reason_pprint_ast.ml

```diff
@@ -3600,13 +3600,13 @@ let printer = object(self:'self)
    * TODO: in the future we should probably use the type ruleCategory
    * to 'automatically' ensure the validity of a constraint expr with parens...
    *)
-  method unparseConstraintExpr ?(ensureExpr=false) e =
+  method unparseProtectedExpr ?(forceParens=false) e =
```

Comment:
The name here "constraint expr" referred to type constraint IIRC. What does "protected" mean?

Reply:
I'll submit a PR with some comments.